File: 刷题/CodeTop/189. 轮转数组.cpp
class Solution {
public:
    void reverse(vector<int>& nums, int l, int r) {
        while(l < r) {
            swap(nums[l], nums[r]);
            l++;
            r--;
        }
    }
    void rotate(vector<int>& nums, int k) {
        reverse(nums, 0, nums.size() - 1);
        reverse(nums, 0, k % nums.size() - 1);
        reverse(nums, k % nums.size(), nums.size() -1 );
    }
};

File: 刷题/Hot 100/189. 轮转数组.go
func reverse(a []int) {
    for i, n := 0, len(a); i < n/2; i++ {
        a[i], a[n-1-i] = a[n-1-i], a[i]
    }
}

func rotate(nums []int, k int) {
    k %= len(nums)
    reverse(nums)
    reverse(nums[:k])
    reverse(nums[k:])
}